### DISCUSSION TITLE
Package oracledb for GUIX

### DISCUSSION THREAD

**Discussion by ennoausberlin** [oracle/python-oracledb#199] [Ideas]
Hello. I am writing a package definition for oracledb for the guix package manager, but I get an error in the "building oracle.base_impl extension phase. cc1 fatal error src/oracledb/base_impl.c No such file or directory. The source package comes from pypi.
Any ideas why this fails and how to build this?

**Comment by anthony-tuininga** [comment]
python-oracledb uses Cython, which means that the Cython source files must be translated into C before the build can take place. This is done by the regular build process. You can use that process so that it generates the C source files or call Cython yourself directly to generate the C source files. After those are generated your can build the regular way.

**Reply by ennoausberlin** [reply]
You put me on the right track. I never used cython before, I just saw gcc failing.  
Anyway, I specified cython as native-input in the GUIX package definition and oracledb was build nicely afterwards. 

Thank you for your quick and profound answer. 
Enno

**Reply by cjbj** [reply]
@ennoausberlin that's great.  Are you making this available to everyone, or is it just for your own use?

**Reply by ennoausberlin** [reply]
Sure. It is just a few lines of code. I had to disable the test phase and will try to re-enable most tests  and publish it then here

**Reply by ennoausberlin** [reply]
```scheme
(define-public python-oracledb
  (package
   (name "python-oracledb")
   (version "1.4.1")
   (source (origin
	    (method url-fetch)
	    (uri (pypi-uri "oracledb" version))
	    (sha256
	     (base32
	      "1brps85jsx7j49g3dj13s5515zgg2zz3d01kcib3fzbc0f0jaqmz"))))
   (build-system pyproject-build-system)
   (arguments '(#:tests? #f))
   (propagated-inputs (list python-cryptography))
   (native-inputs (list python-pytest python-cython-3))
   (home-page "https://oracle.github.io/python-oracledb")
   (synopsis "Python interface to Oracle Database")
   (description "Python interface to Oracle Database")
   (license license:asl2.0)))
```
